caas/kubernetes/provider/application/application.go

```diff
@@ -875,18 +885,23 @@ func (a *app) applicationPodSpec(config caas.ApplicationConfig) (*corev1.PodSpec
 
 func (a *app) annotations(config caas.ApplicationConfig) annotations.Annotation {
 	return k8sutils.ResourceTagsToAnnotations(config.ResourceTags).
-		Add(constants.LabelVersion, config.AgentVersion.String())
+		Merge(k8sutils.AnnotationsForVersion(config.AgentVersion.String(), a.legacyLabels))
 }
 
 func (a *app) labels() map[string]string {
 	// TODO: add modelUUID for global resources?
-	return map[string]string{constants.LabelApplication: a.name}
+	return k8sutils.LabelsForApp(a.name, a.legacyLabels)
+}
+
+func (a *app) selectorLabels() map[string]string {
```

Comment:
It may be worth changing this function signature to return labels.Set

Reply:
done